examples/nft-nd/nft.go

```diff
@@ -285,3 +287,63 @@ func Properties(id []byte) map[string]string {
 	}
 	return result
 }
+
+// RoyaltyRecipient contains information about the recipient and the royalty amount.
+type RoyaltyRecipient struct {
+	Address interop.Hash160
+	Amount  int
+}
+
+// RoyaltyInfo returns a list of royalty recipients and the corresponding royalty amounts.
+func RoyaltyInfo(tokenID []byte, royaltyToken interop.Hash160, salePrice int) []RoyaltyRecipient {
+	if salePrice <= 0 {
+		panic("sale price must be positive")
+	}
+
+	callingHash := runtime.GetExecutingScriptHash()
+	if !royaltyToken.Equals(callingHash) {
+		panic("invalid royalty token")
+	}
+
+	ctx := storage.GetReadOnlyContext()
+	owner := getOwnerOf(ctx, tokenID)
+	if owner == nil {
+		panic("invalid token ID")
+	}
+
+	royaltyInfoKey := append([]byte(royaltyInfoPrefix), tokenID...)
+	val := storage.Get(ctx, royaltyInfoKey)
+
+	var recipients []RoyaltyRecipient
+	if val == nil {
+		recipients = []RoyaltyRecipient{
+			{
+				Address: owner,
+				Amount:  (salePrice * 10) / 100,
+			},
+		}
+	} else {
+		storedRecipients := val.([]any)
+		for _, r := range storedRecipients {
+			recipient := r.([]any)
+			recipients = append(recipients, RoyaltyRecipient{
+				Address: recipient[0].(interop.Hash160),
+				Amount:  recipient[1].(int),
```

Comment:
> recipient[1].(int),

It's not the amount; the contract stores the *share rate* for a particular address.